src/script/lua_api/l_object.cpp

```diff
@@ -235,8 +235,14 @@ int ObjectRef::l_set_hp(lua_State *L)
 		return 0;
 
 	// Get HP
+	int old_hp = co->getHP();
 	int hp = lua_tonumber(L, 2);
 
+	// Protect from damage if immortal
+	if ((itemgroup_get(co->getArmorGroups(), "immortal") == 1) && (hp < old_hp)) {
```

Comment:
still doesn't use `co->isImmortal()`

Reply:
Also, shouldn't this be in RemotePlayer/PlayerSAO instead of the script api?

Reply:
Oops :)

Yeah, I'll move this to PlayerSAO. I don't think the `hp < old_hp` check is required either - if an player is immortal, their HP shouldn't increase either. Or... should it?

Reply:
Nvm, the exact code already exists in `PlayerSAO::setHP` since a687efa6